### PR TITLE
refactor(text-input): migrate to TypeScript

### PR DESCRIPTION
**Pull request by adnasa**
#### Summary

- blocked by #1867
- This is cherry picked work, so I expect that the build will fail until the blocking PR is merged

Assigning review request by random.

### CODE REVIEW

File: packages/components/inputs/text-input/src/text-input.tsx [hunk 1]
```diff
@@ -1,127 +1,131 @@
...
-        contentEditable={!props.isReadOnly}
-      />
-    </Constraints.Horizontal>
-  );
-};
-
-TextInput.displayName = 'TextInput';
-
-TextInput.propTypes = {
+type TTextInputProps = {
```
Comment (fuchodeveloper) [inline packages/components/inputs/text-input/src/text-input.tsx:7]: Do we have a convention typing convention for ui-kit at the moment?

Comment (adnasa) [reply packages/components/inputs/text-input/src/text-input.tsx:7]: thanks for this question, but is there something specific you're referring to?
I set a `T*` prefix to differentiate Types and Interface for devs who rely on editors without intellisense. Does that make sense?

Comment (fuchodeveloper) [reply packages/components/inputs/text-input/src/text-input.tsx:7]: Basically, I'm asking if using a naming like, `T*` prefix, is a convention for ui-kit. This potentially could help with migrations for other files and avoids ambiguity.
Does that help?

Comment (adnasa) [reply packages/components/inputs/text-input/src/text-input.tsx:7]: then yes it is

**Review by fuchodeveloper — APPROVED** [review]

**Review by dogayuksel — APPROVED** [review]

**Pull request merged**